modules/mono/glue/GodotSharp/GodotSharp/Core/NativeInterop/Marshaling.cs

```diff
@@ -411,53 +371,46 @@ private static unsafe godot_variant mono_object_to_variant_impl(object p_obj, bo
                             foreach (KeyValuePair<object, object> entry in (IDictionary)p_obj)
                                 godotDict.Add(entry.Key, entry.Value);
 
-                            return VariantUtils.CreateFromDictionary(godotDict.NativeValue);
+                            return VariantUtils.CreateFromDictionary((godot_dictionary)godotDict.NativeValue);
                         }
 
                         if (genericTypeDefinition == typeof(System.Collections.Generic.List<>))
                         {
                             // TODO: Validate element type is compatible with Variant
-                            var nativeGodotArray = mono_array_to_Array((IList)p_obj);
-                            return VariantUtils.CreateFromArray(&nativeGodotArray);
+                            using var nativeGodotArray = ConvertIListToNativeGodotArray((IList)p_obj);
+                            return VariantUtils.CreateFromArray(nativeGodotArray);
                         }
                     }
 
                     break;
                 }
             }
 
-            if (p_fail_with_err)
-            {
-                GD.PushError("Attempted to convert an unmarshallable managed type to Variant. Name: '" +
-                             p_obj.GetType().FullName + ".");
-                return new godot_variant();
-            }
-            else
-            {
-                return new godot_variant();
-            }
+            GD.PushError("Attempted to convert an unmarshallable managed type to Variant. Name: '" +
+                         p_obj.GetType().FullName + ".");
+            return new godot_variant();
         }
 
-        public static unsafe string variant_to_mono_string(godot_variant* p_var)
+        private static string? variant_to_mono_string(in godot_variant p_var)
```

Comment:
Did you forget to rename this method? Since you renamed all the others.

Reply:
It seems so, although it's not too important. The public method names are the ones I wanted to sort out.